lib/teleterm/api/proto/v1/cluster.proto

```diff
@@ -22,15 +22,21 @@ option go_package = "github.com/gravitational/teleport/lib/teleterm/v1";
 message Cluster {
     // uri is the cluster resource URI
     string uri = 1;
-    // name is the cluster name
+    // name is the host/profile name
```

Comment:
I'd explain the situation in more detail, as I suspect just from reading those comments alone it might not be clear for members of other teams what the issue is.

```suggestion
    // name is used throughout the Teleport Connect codebase as the cluster name. Unfortunately, we made a mistake where instead of taking the name of the cluster from the key, we take it from the hostname part of the proxy service. These two are not always equal. And actually there are places in tsh code where it needs the name of the cluster from the key, not the hostname of the proxy service (like in the TELEPORT_CLUSTER env var).
    // Since this identifier is so widely used, we didn't have time to change it before the preview release. In the future we plan to fix the value sent under name to the name from the key and get rid of the `actual_name` field.
```